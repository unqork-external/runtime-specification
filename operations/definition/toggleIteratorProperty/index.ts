export * from './toggleIteratorProperty.model'
export * from './toggleIteratorProperty.options'
