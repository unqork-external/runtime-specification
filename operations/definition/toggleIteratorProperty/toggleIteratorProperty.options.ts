import { Default, Description, Optional } from '@tsed/schema'

export class ToggleIteratorPropertyOperationOptions {
  @Description('Key of the component to expand/collapse.')
  targetKey: string

  @Description('The row path inside the `targetKey` to a node to expand/collapse.')
  rowTargetPath?: string

  @Description('Which property we want to toggle.')
  toggleProperty: string

  @Optional()
  @Description('What we should set the toggleProperty to. If no value is set, we flip it.')
  toggleValue?: boolean

  @Description(
    'If true, apply Op to all children of the target node and expand/collapse any children from this node down.',
  )
  @Default(false)
  applyToChildren: boolean = false
}
