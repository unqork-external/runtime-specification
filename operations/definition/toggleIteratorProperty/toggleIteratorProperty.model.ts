import { DiscriminatorValue, Required } from '@tsed/schema'

import { ToggleIteratorPropertyOperationOptions } from './toggleIteratorProperty.options'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.TOGGLE_ITERATOR_PROPERTY)
export class ToggleIteratorPropertyOperation extends Operation<OperationTypes.TOGGLE_ITERATOR_PROPERTY> {
  @Required()
  public type = OperationTypes.TOGGLE_ITERATOR_PROPERTY as const

  @Required()
  public options: ToggleIteratorPropertyOperationOptions = new ToggleIteratorPropertyOperationOptions()
}
