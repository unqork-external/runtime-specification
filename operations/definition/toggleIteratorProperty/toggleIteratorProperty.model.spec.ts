import { ToggleIteratorPropertyOperation } from './toggleIteratorProperty.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: ToggleIteratorProperty Tests', () => {
  let schema
  beforeAll(() => {
    const response = generateSchemaAndValidate(ToggleIteratorPropertyOperation)
    schema = response.schema
  })

  it('should match this JSON Schema', () => {
    expect(schema).toMatchSnapshot()
  })
})
