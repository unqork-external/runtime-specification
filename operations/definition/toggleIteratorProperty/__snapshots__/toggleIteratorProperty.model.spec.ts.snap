// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: ToggleIteratorProperty Tests should match this JSON Schema 1`] = `
{
  "definitions": {
    "ToggleIteratorPropertyOperationOptions": {
      "properties": {
        "applyToChildren": {
          "default": false,
          "description": "If true, apply Op to all children of the target node and expand/collapse any children from this node down.",
          "type": "boolean",
        },
        "rowTargetPath": {
          "description": "The row path inside the \`targetKey\` to a node to expand/collapse.",
          "type": "string",
        },
        "targetKey": {
          "description": "Key of the component to expand/collapse.",
          "type": "string",
        },
        "toggleProperty": {
          "description": "Which property we want to toggle.",
          "type": "string",
        },
        "toggleValue": {
          "description": "What we should set the toggleProperty to. If no value is set, we flip it.",
          "type": "boolean",
        },
      },
      "type": "object",
    },
  },
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/ToggleIteratorPropertyOperationOptions",
    },
    "type": {
      "const": "TOGGLE_ITERATOR_PROPERTY",
      "examples": [
        "TOGGLE_ITERATOR_PROPERTY",
      ],
      "type": "object",
    },
  },
  "required": [
    "type",
    "options",
  ],
  "type": "object",
}
`;
